napari/utils/events/containers/_selectable_list.py

```diff
@@ -77,8 +77,14 @@ def remove_selected(self):
         for i in list(self.selection):
             idx = self.index(i)
             self.remove(i)
-        new = max(0, (idx - 1))
-        if len(self) > new:
+        if isinstance(idx, int):
+            new = max(0, (idx - 1))
+            do_add = len(self) > new
+        else:
+            *root, _idx = idx
+            new = tuple(root) + (_idx - 1,) if _idx >= 1 else tuple(root)
+            do_add = len(self) > new[0]
+        if do_add:
```

Comment:
yeah, seems like it kinda picks a random index.  The integer based logic came from the early layer list stuff I think, the tuple stuff here is just added for nestle layer lists.  I think the goal is just to make sure that something adjacent to one of the selected items in the list is selected after deletion.  Since that behavior has been there forever, can we address that in a follow up PR if you find the behavior weird?

Reply:
hm. 
My first worry is that it is a little weird that the selected contains a mixed representation of the index (`1` and (`(1,)`). But I see that there are also some profits. 

As I understand the idea  I think that some better logic should be applied as the previous logic was to select a layer before last removed. So logic needs to be a little more complicated. Maybe it is enough to do separate PR. But if you could add a comment with an explanation of what is expected (especially if you do not have time to do this in a short time after this PR).

Reply:
> My first worry is that it is a little weird that the selected contains a mixed representation of the index (1 and ((1,))

yes that's going to result from nested indices.  also not related to this PR though...